Harmonize layer creation description (#4010)
qgis/QGIS-Documentation: source/docs/user_manual/managing_data_source/create_layers.rst

Creating new vector layers
==========================

QGIS allows you to create new layers in different formats. It provides tools

> for creation from scratch for GeoPackage, Shapefile, SpatiaLite, GPX and

Reviewer: ```suggestion
for creation from scratch for GeoPackage, Shapefile, SpatiaLite, GPX formats and
```

> Temporary Scratch layers (aka memory layers) formats. Creation of a :ref:`new

Reviewer: ```suggestion
Temporary Scratch layers (aka memory layers). Creation of a :ref:`new
```

   Creating a New GeoPackage layer dialog

#. The first step is to indicate the database file location. This can be done
   by pressing the :guilabel:`...` button at the right of the
   :guilabel:`Database` field and select an existing GeoPackage file
   or create a new one. QGIS will automatically add the right extension to

>    the name you provide

Reviewer: ```suggestion
   the name you provide.
```

#. Then, give a name for the new layer to store
#. Define the :guilabel:`Geometry type`. If not a geometryless layer, you

>    can add whether it should :guilabel:`Include Z dimension` and/or

Reviewer: ```suggestion
   can specify if it should :guilabel:`Include Z dimension` and/or
```

>    :guilabel:`Include M values`

Reviewer: ```suggestion
   :guilabel:`Include M values`.
```

> #. Specify the coordinate reference system using the |setProjection| button.

Reviewer: ```suggestion
#. Specify the coordinate reference system using the |setProjection| button
```

To add fields to the layer you are creating:

#. Enter the :guilabel:`Name` of the field
#. Select the data :guilabel:`Type`. Supported types are :guilabel:`Text data`,
   :guilabel:`Whole number` (both integer and integer64), :guilabel:`Decimal
   number`, :guilabel:`Date` and :guilabel:`Date and time`, :guilabel:`Blob`

>    and :guilabel:`Boolean`

Reviewer: ```suggestion
   and :guilabel:`Boolean`.
```

#. Depending on the selected data format, enter the :guilabel:`Maximum length`

>    of values

Reviewer: ```suggestion
   of values.
```

> #. Click on the |newAttribute| :guilabel:`Add to Fields List` button.

Reviewer: ```suggestion
#. Click on the |newAttribute| :guilabel:`Add to Fields List` button
```

#. And reproduce the steps above for each field you need to add.
#. Once you are happy with the attributes, click :guilabel:`OK`.
   QGIS will automatically add the new layer to the legend, and you can edit it
   in the same way as described in section :ref:`sec_edit_existing_layer`.

By default, When creating a GeoPackage layer, QGIS generates a :guilabel:`Feature
id column` called ``fid`` which behaves like the primary key of the layer. The
name can be changed. Likewise, the geometry field, if availabe, is named ``geometry``

> and you can enable to :guilabel:`Create a spatial index` on it.

Reviewer: ```suggestion
and you can specify to :guilabel:`Create a spatial index` on it.
```

These options can be found under the :guilabel:`Advanced Options` next to the
:guilabel:`Layer identifier`, short human readable name of the layer, and the
:guilabel:`Layer description`.

Further management of GeoPackage layers can be done with the :ref:`DB Manager
<dbmanager>`.

.. _vector_create_shapefile:

Creating a new Shapefile layer
------------------------------

To create a new Shapefile layer, press the |newVectorLayer| :menuselection:`New

> Shapefile Layer...` button from the :menuselection:`Layer --> Create Layer -->`

Reviewer: ```suggestion
Shapefile Layer...` button in the :menuselection:`Layer --> Create Layer -->`
```

menu or from the :guilabel:`Data Source Manager` toolbar.
The :guilabel:`New Shapefile Layer` dialog will be displayed as shown in
figure_create_shapefile_.

#. The first step is to provide a path and name for the layer using the
   :guilabel:`...` button next to :guilabel:`File name`. QGIS will

>    automatically add the right extension to the name you provide

Reviewer: ```suggestion
   automatically add the right extension to the name you provide.
```

#. Next, indicate the :guilabel:`File encoding` of the data

> #. Choose the :guilabel:`Geometry type` of layer (point, multipoint,line or polygon)

Reviewer: ```suggestion
#. Choose the :guilabel:`Geometry type` of the layer (point, multipoint,line or polygon)
```

> #. Check whether the geometry has an optional :guilabel:`Z (+ M values)` or

Reviewer: ```suggestion
#. Check whether the geometry should have an optional :guilabel:`Z (+ M values)` or
```

   Creating a new Shapefile layer dialog

To add fields to the layer you are creating:

#. Enter the :guilabel:`Name` of the field
#. Select the data :guilabel:`Type`. Only :guilabel:`Decimal number`,

>    :guilabel:`Whole number`, :guilabel:`Text data` and :guilabel:`Date` formats

Reviewer: ```suggestion
   :guilabel:`Whole number`, :guilabel:`Text data` and :guilabel:`Date`
```

>    attributes are supported

Reviewer: ```suggestion
   attributes are supported.
```

#. Depending on the selected data format, enter the :guilabel:`Length` and
   :guilabel:`Precision`.

> #. Click on the |newAttribute| :guilabel:`Add to Fields List` button.

Reviewer: ```suggestion
#. Click on the |newAttribute| :guilabel:`Add to Fields List` button
```

> #. And reproduce the steps above for each field you need to add.

Reviewer: ```suggestion
#. And reproduce the steps above for each field you need to add
```

#. Once you are happy with the attributes, click :guilabel:`OK`.
   QGIS will automatically add the new layer to the legend, and you can edit it

>    in the same way as described in section :ref:`sec_edit_existing_layer`.

Reviewer: ```suggestion
   as described in section :ref:`sec_edit_existing_layer`.
```

> By default, A first integer ``id`` column is added but can be removed, if not

Reviewer: ```suggestion
By default, A first integer ``id`` column is added but can be removed.
```

> desired.

Reviewer: ```suggestion
```

   Creating a New SpatiaLite layer dialog

#. The first step is to indicate the database file location. This can be done
   by pressing the :guilabel:`...` button at the right of the
   :guilabel:`Database` field and select an existing SpatiaLite file
   or create a new one. QGIS will automatically add the right extension to

>    the name you provide

Reviewer: ```suggestion
   the name you provide.
```

#. Then, give a name for the new layer to store
#. Define the :guilabel:`Geometry type`. If not a geometryless layer, you

>    can add whether it should :guilabel:`Include Z dimension` and/or

Reviewer: ```suggestion
   can specify whether it should :guilabel:`Include Z dimension` and/or
```

>    :guilabel:`Include M values`

Reviewer: ```suggestion
   :guilabel:`Include M values`.
```

#. Specify the coordinate reference system using the |setProjection| button.

To add fields to the layer you are creating:

#. Enter the :guilabel:`Name` of the field
#. Select the data :guilabel:`Type`. Supported types are :guilabel:`Text data`,

>    :guilabel:`Whole number` and :guilabel:`Decimal number`

Reviewer: ```suggestion
   :guilabel:`Whole number` and :guilabel:`Decimal number`.
```

> #. Click on the |newAttribute| :guilabel:`Add to Fields List` button.

Reviewer: ```suggestion
#. Click on the |newAttribute| :guilabel:`Add to Fields List` button
```

> #. And reproduce the steps above for each field you need to add.

Reviewer: ```suggestion
#. And reproduce the steps above for each field you need to add
```

#. Once you are happy with the attributes, click :guilabel:`OK`.
   QGIS will automatically add the new layer to the legend, and you can edit it

>    in the same way as described in section :ref:`sec_edit_existing_layer`.

Reviewer: ```suggestion
   as described in section :ref:`sec_edit_existing_layer`.
```
